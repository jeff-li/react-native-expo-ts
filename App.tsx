import { StatusBar } from 'expo-status-bar';
import { NativeBaseProvider } from 'native-base';
import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import useCachedResources from './hooks/useCachedResources';
import useColorScheme from './hooks/useColorScheme';
import Navigation from './navigation';
import { AuthContextProvider } from './contexts/AuthContext'

export default function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      <SafeAreaProvider>
        <NativeBaseProvider>
          <AuthContextProvider>
            <Navigation data-testid="nav" colorScheme={colorScheme} />
            <StatusBar />
          </AuthContextProvider>
        </NativeBaseProvider>
      </SafeAreaProvider>
    );
  }
}
