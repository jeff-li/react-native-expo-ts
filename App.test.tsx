import React from "react";
import { mocked } from 'ts-jest/utils';
import { render } from '@testing-library/react-native';
import useCachedResources from './hooks/useCachedResources';

import App from "./App";

const mockedCachedResources = mocked(useCachedResources);
jest.mock('./hooks/useCachedResources', () => {
  return jest.fn(() => true)
})

jest.mock('./hooks/useColorScheme', () => {
  return jest.fn(() => 'dark')
})

describe("<App />", () => {
  it("should match screenshot if loading is complete", () => {
    mockedCachedResources.mockReturnValue(true)
    const { toJSON } = render(
      <App />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("should return undefined if app is still loading", () => {
    mockedCachedResources.mockReturnValue(false)
    const { toJSON } = render(
      <App />
    );
    expect(toJSON()).toBe(null);
  });
});