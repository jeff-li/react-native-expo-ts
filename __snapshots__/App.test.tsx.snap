// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match screenshot if loading is complete 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
